my-vault/internal/handlers/handlers.go

```diff
@@ -87,77 +82,75 @@ func HandlePostSecret(w http.ResponseWriter, r *http.Request) {
 		http.Error(w, err.Error(), http.StatusBadRequest)
 		return
 	}
-
+	// build new secretObject
 	temp := NewCreateSecret(secretRequestObject)
-	mu.Lock()
-	db[temp.Id] = temp
-	mu.Unlock()
-	
+	err = db.PutNewSecret(temp)
+
+	if err != nil {
+		
+		err := errors.New("unexpected internal error")
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
 	res := CreateSecretResponse{
 		Id:      temp.Id,
 		Name:    temp.Name,
 		Message: fmt.Sprintf("Successfully created secret with secretId: %s and name: %s", temp.Id, temp.Name),
 	}
-
+	fmt.Printf("Secret saved----------------------------------------- %s", temp.Id)
 	w.Header().Set("Content-Type", "application/json")
 	encoder := json.NewEncoder(w)
 	err = encoder.Encode(res)
-
 	if err != nil {
 		
 		err := errors.New("unexpected internal error")
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 		return
 	}
-
-	fmt.Printf("Secret saved----------------------------------------- %s", secret)
 }
 
 func HandleGetSecretById(w http.ResponseWriter, r *http.Request) {
 	//curl  -X POST -H "Content-Type: application/json" http://localhost:9000/secrets/234
 	secretID := r.PathValue("secret_id")
-	mu.Lock()
-	secretEntry, ok := db[secretID]
-	mu.Unlock()
 	
-	if !ok {
+	secretEntry,err := db.GetScretsById(secretID)
+	
+	if err!=nil {
 		http.Error(w, fmt.Sprintf("Secret with ID %s not found", secretID), http.StatusNotFound)
 		return
 	}
+
 	w.Header().Set("Content-Type", "application/json")
 	getSecretResponse := NewSecretResponseObject(secretEntry)
 	json.NewEncoder(w).Encode(getSecretResponse)
 	fmt.Printf("Retrieved secret----------------------------------------- %s", getSecretResponse)
 }
 
+
 func HandleDeleteSecretById(w http.ResponseWriter, r *http.Request) {
 	//curl  -X POST -H "Content-Type: application/json" http://localhost:9000/secrets/234
-	secret_id := r.PathValue("secret_id")
-	if _, exists := db[secret_id]; exists {
-		mu.Lock()
-		delete(db, secret_id)
-		mu.Unlock()
-		fmt.Println("Deleted secret with id -----------------------------------------", secret_id)
-		fmt.Fprintf(w, "Secret with ID %s deleted successfully", secret_id)
-	} else {
-		http.Error(w, fmt.Sprintf("Secret with ID %s not found", secret_id), http.StatusNotFound)
+	secretId := r.PathValue("secret_id")
+	secret,_ := db.DeleteSecretByID(secretId)
+
+	if secret != "" {
+		fmt.Println("Deleted secret with id -----------------------------------------", secret)
+		fmt.Fprintf(w, "Secret with ID %s deleted successfully", secret)
+
+	}else {
+		http.Error(w, fmt.Sprintf("Secret with ID %s not found", secret), http.StatusNotFound)
 		return
 	}
 
 }
 
 func HandleListSecrets(w http.ResponseWriter, r *http.Request) {
-	var secrets []Secret
-	mu.Lock()
-	if len(db) == 0 {
+
+	secrets, err := db.ListAllSecrets()
+	if err!=nil{
 		http.Error(w, "No secrets found in the secrets store", http.StatusNotFound)
```

Comment:
Is this the only error that can happen when listing? I wouldn't assume that at this level.